### PR TITLE
Merge repair and rearm anim traits into WithResupplyAnimation

### PR DESCRIPTION
**Pull request by reaperrr**
Another (and definitely the last) split from #14357.

This is the safest approach to avoid animation conflicts/visual glitches when the 'host' is responsible for both resupply types.
The new `WithResupplyAnimation` trait will simply play a looping animation as long as the actor is resupplying in any form.

As a byproduct, this makes the repair/reload animations look more like in the originals and less erratic.
We don't have a `WithRearmOverlay` yet, so I'll leave a refactor of `WithRepairOverlay` for later.

### CODE REVIEW

**Comment by reaperrr** [comment]
@Mailaender regarding your comment on IRC, I'll add some context:

This was split from a larger PR.

Basically, the ultimate goal of this PR, #14357 which it was split from and the PR(s) that will follow, is to allow actors to be reloaded and repaired _at the same time_ (right now on bleed, actors that need both will be rearmed first, then repaired, as both are separate activities that are queued in `Repairable`).

The problem is that on bleed, the reload/repair animation (of the building) is played once on every repair/reload step, instead of just looping continuously. 
So when repairing and rearming happen in parallel once the refactoring is done, depending on their interval it's possible that, for example, a repair step happens just a tick after a rearm step, resulting in restarting the service depot animation before the last one finished.
This is also already an issue with rearming or repairing when the resupply interval is shorter than the animation, as in that case it restarts prematurely as well.

This PR simply makes the resupply animations of helipads, service depots and so on loop at consistent speed as long as the building is either repairing or rearming (or both) some actor.

By the way, this also makes the animations look more like in the original, where the animation speed wasn't tied to the repair/reload speed, either, making it look less hectic.

**Comment by Mailaender** [comment]
If a vehicle gets killed while repairing/rearming the animation continues:

image

I don't know if that is a regression or already existing.

**Review by Mailaender — DISMISSED** [review]
Rearming and repairing still works and the animation plays.

**Comment by reaperrr** [comment]
> If a vehicle gets killed while repairing/rearming the animation continues

Should be an easy fix, if that's the case I might as well include it here.

**Comment by reaperrr** [comment]
Rebased onto #15622 and fixed the above issue.

**Comment by reaperrr** [comment]
Rebased onto updated #15622 and fixed that ResupplyAircraft only notified INotifyRepair (unconditionally).
It now checks which of the activities was running on disposal and then notifies the correct trait(s) accordingly.

**Comment by reaperrr** [comment]
Rebased and updated again.
The update only affects `Rearm` activity though, it now properly gets the host passed like `Repair` instead of relying on the old ActorsAt hacks.

According to my own tests everything still works as it should, but I guess there have been too many changes since @Mailaender's review to keep his +1.

**Review by abcdefg30 — DISMISSED** [review]
Looks fine otherwise and ingame. :+1: 
Do we want to move the update rule into a folder like in #14664?

File: OpenRA.Mods.Common/Traits/Render/WithResupplyAnimation.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	public class WithResupplyAnimationInfo : ConditionalTraitInfo, Requires<WithSpriteBodyInfo>
+	{
+		[Desc("Sequence name to use")]
+		[SequenceReference] public readonly string Sequence = "active";
+
+		[Desc("Which sprite body to play the animation on.")]
+		public readonly string Body = "body";
+
+		[Desc("Events leading to the animation getting played. Possible values currently are: Rearm, Repair.")]
+		public readonly ResupplyType PlayAnimationOn = ResupplyType.Rearm | ResupplyType.Repair;
```
[the remark this answers is not in the document]
Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/Render/WithResupplyAnimation.cs]: Not sure, tbh. Removed, we can always add it back later if needed, afterall.

**Comment by reaperrr** [comment]
Updated.

> Do we want to move the update rule into a folder like in #14664?

Done.

**Comment by reaperrr** [comment]
Rebased.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d5913ed49be02ef26ec6a772051f6b87baade90e...b8a3afc425a2588092f098e49b9e0f251723972e)

**Pull request merged**